src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,28 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_AGE = 18;
+    private static final int MINIMUM_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getAge() == null || user.getAge() < MINIMUM_AGE) {
+            throw new RuntimeException("Wrong age");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < MINIMUM_LENGTH) {
+            throw new RuntimeException("Wrong login");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < MINIMUM_LENGTH) {
+            throw new RuntimeException("Wrong password");
+        }
+        if (isStorageContainLogin(user.getLogin())) {
+            throw new RuntimeException("Such user exist in DB");
```

Comment:
The exception message 'Such user exist in DB' could be more informative. For example, 'User with login ' + user.getLogin() + ' already exists in the database.'